flow/aggregates.py

```diff
@@ -0,0 +1,498 @@
+    """This class holds the information of the project associated with
+    an operation function aggregated by the default aggregates i.e.
+    :py:class:`aggregator.groupsof(1)`.
+
+    When iterated through it's instance, it yields a tuple of a single job from
+    the Project.
+
+    :param project:
+        A signac project used to fetch jobs for creating aggregates.
+    :type project:
+        :py:class:`flow.FlowProject` or :py:class:`signac.contrib.project.Project`
+    """
+    def __init__(self, project):
+        self._project = project
+
+    def __iter__(self):
+        for job in self._project:
+            yield (job,)
+
+    def __getitem__(self, id):
+        "Return a tuple of a single job via job id."
+        try:
+            return (self._project.open_job(id=id),)
+        except KeyError:
+            raise LookupError(f"Did not find job with id {id}.")
+
+    def __contains__(self, job):
+        """Return whether the job is present in the project associated with this
+        instance of :py:class:`_DefaultAggregateStore`.
+        """
+        return job in self._project
+
+    def __len__(self):
+        return len(self._project)
+
+    def __eq__(self, other):
+        return type(self) == type(other) and \
+               self._project == other._project
+
+    def __hash__(self):
+        blob = self._project.id
+        return int(sha1(blob.encode('utf-8')).hexdigest(), 16)
+
+    def _register_aggregates(self, project):
+        """We have to store self._project when this method is invoked
+        This is because we will then iterate over that project in
+        order to return an aggregates of one.
+        """
+        self._project = project
+
+
+class _MakeAggregates:
+    """This class acts as a funnctor which handles the creation of
+    aggregates using all the jobs in a project.
+
+    :param aggregator_function:
+        The ``aggregator_function`` attribute associated with the
+        :py:class:`aggregator`
+    :type aggregator_function:
+        callable
+    :param sort_by:
+        The ``sort_by`` attribute associated with the :py:class:`aggregator`
+    :type sort_by:
+        str or NoneType
+    :param reverse_order:
+        The ``reverse_order`` attribute associated with the :py:class:`aggregator`
+    :type reverse_order:
+        bool
+    :param select:
+        The ``select`` attribute associated with the :py:class:`aggregator`
+    :type select:
+        callable or NoneType
+    :param project:
+        A signac project used to fetch jobs for creating aggregates.
+    :type project:
+        :py:class:`flow.FlowProject` or :py:class:`signac.contrib.project.Project`
+    """
+    def __init__(self, aggregator_function, sort_by, reverse_order, select, project):
+        self._aggregator_function = aggregator_function
+        self._sort_by = sort_by
+        self._reverse_order = bool(reverse_order)
+        self._select = select
+        self._project = project
```

Comment:
I don't know why we are storing the project in this class. I know I missed this last review. I believe passing in the project to `_MakeAggregates` is the proper approach. That provides the intuitive behavior of `_MakeAggregates` as a function that takes in a project and returns aggregates.